src/python/pants/bsp/util_rules/targets.py

```diff
@@ -225,32 +258,21 @@ async def generate_one_bsp_build_target_request(
     metadata = metadata_results_by_lang_id[metadata_merge_order[-1].language_id].metadata
     digest = await Get(Digest, MergeDigests([r.digest for r in metadata_results]))
 
-    # Determine base directory for this build target using source roots.
-    targets_with_sources = [tgt for tgt in targets if tgt.has_field(SourcesField)]
-    sources_paths = await MultiGet(
-        Get(SourcesPaths, SourcesPathsRequest(tgt[SourcesField])) for tgt in targets_with_sources
-    )
-    _logger.info(f"sources_paths = {sources_paths}")
-    merged_source_files: set[str] = set()
-    for sp in sources_paths:
-        merged_source_files.update(sp.files)
-
-    source_roots_result = await Get(
-        SourceRootsResult, SourceRootsRequest, SourceRootsRequest.for_files(merged_source_files)
-    )
-    source_root_paths = {x.path for x in source_roots_result.path_to_root.values()}
-    if len(source_root_paths) == 0:
-        base_dir = build_root.pathlib_path
-    elif len(source_root_paths) == 1:
-        base_dir = build_root.pathlib_path.joinpath(list(source_root_paths)[0])
+    # Determine "base directory" for this build target using source roots.
+    # TODO: This actually has nothing to do with source roots. It should probably be computed as an ancestor
+    # directory or else be configurable by the user. It is used as a hint in IntelliJ for where to place the
+    # corresponding IntelliJ module.
```

Comment:
Should this comment be on the class instead?

Reply:
I'm going to make it configurable in another PR, so going to leave as-is for now.